### DISCUSSION TITLE
Don't suppress the focalPoint attribute for non-image assets

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#10415] [Ideas] [answered]
Currently, [Craft suppresses the `focalPoint` attribute for non-image assets](https://github.com/craftcms/cms/blob/b4ea1625c9ca75f3502f6dfbb7e219bd90cb3624/src/elements/Asset.php#L1655) – i.e. `null` is returned, even if the `focalPoint` database column has a value). This is similar to how _dimensions_ used to be suppressed for non-image assets as well, until that changed in 3.7.21 as a result of #9038.

Even if Craft itself doesn't currently expose a way to set the focal point for non-image assets (for example, videos) inside the CP, I think not suppressing the database value (i.e. if it was set by a plugin/module) has merit.

A concrete use case, I'm sure there are more: 

We've built a headless, bespoke DAM app on top of Craft. The app lets users edit images and videos, and includes a custom interface to set focal points for images. We'd like to add the ability to set focal points for videos as well (primarily to use the focal point values as basis for the `object-position` CSS property, when embedding videos stored in the DAM). The obvious workaround is to use a custom field for the videos' focal points, but it'd definitely be nicer to just be able to use the native attribute.

**Comment by brandonkelly** [comment]
738906772dc only opened up the dimensions to videos, not all file types. Would opening focal points up only to videos work here as well, or are you really hoping for all file types?

**Reply by mmikkel** [reply]
You're correct, and my specific use case involves videos – so yeah, opening up focal points to videos only would suffice :)

**Comment by brandonkelly** [comment, marked as answer]
Resolved for the next release (59c7427579b2e45f8b953a75f37d86aa945bcde5).

**Reply by mmikkel** [reply]
Awesome, thanks Brandon!

**Reply by brandonkelly** [reply]
3.7.31 is out now with that change.
